include/avk/semaphore.hpp

```diff
@@ -48,6 +48,36 @@ namespace avk
 		const auto& handle() const { return mSemaphore.get(); }
 		const auto* handle_addr() const { return &mSemaphore.get(); }
 
+		// timeline semaphore specific functions
+
+		/** @brief returns the current value of the timeline semaphore */
+		const uint64_t query_current_value() const;
+
+		/** @brief sets the timeline semaphore to the specified value */
+		void signal(uint64_t aNewValue) const;
+
+		/** @brief Waits on host until the timiline semaphore reaches the given value or the timeout(in nanoseconds) happens.
+		*   @return Value of type vk::Result containing information about whether the wait operation succeeded, or the timeout has been triggered.
+		*/
+		vk::Result wait_until_signaled(uint64_t aRequiredValue, std::optional<uint64_t> aTimeout = {}) const;
+
+		/** @brief Waits on host until the condition specified with the parameters is met.
+		*   @param aSemaphores Vector of timeline semaphores that should be waited on. All semaphores are required to be owned by the same logical device.
+		*   @param aTimestamps Vector of payload values to wait on. Is required to have the same size as aSemaphores. The n-th value in aTimestamps corresponds to the n-th entry in aSemaphores.
+		*   @param aWaitOnAll (optional) If true, waits until ALL semaphores have reached their target timestamps. If false, waits until ANY semaphore has reached its target timestamp.
+		*   @param aTimeout (optional) Defines a timeout (in nanoseconds) after which the function returns regardless of the semaphore state.
+		*   @return Value of type vk::Result containing information about whether the wait operation succeeded, or the timeout has been triggered.
+		*/
+		static vk::Result wait_until_signaled(const std::span<std::reference_wrapper<const semaphore_t>> aSemaphores, const std::span<uint64_t> &aTimestamps, bool aWaitOnAll = true, std::optional<uint64_t> aTimeout = {});

```

Comment:
This is somewhat out of place here. Better turn it into a free function and move it somewhere into `class root`!

Reply:
Usage of `std::span` is wrong here. `std::span` should never be passed as `const&`.